packages/core/src/auth/providers/credentialsProviderManager.ts

```diff
@@ -43,8 +51,14 @@ export class CredentialsProviderManager {
             if (!providerType) {
                 continue
             }
-            const telemType = credentialsProviderToTelemetryType(providerType)
-            telemetry.aws_loadCredentials.emit({ credentialSourceId: telemType, value: refreshed.length })
+            const emitWithDebounce = cancellableDebounce(
```

Comment:
I don't think this will work as you intend. I think you need to move this cancellableDebounce() call to a class parameter since this creates a new instance each time